### PR TITLE
(feat) Add auto-discovery in k8s | Adarsh

### PR DESCRIPTION
**Pull request by adarshk-rapido**

### CODE REVIEW

**Comment by adarshk-rapido** [comment]
will do a bit more testing from my end and reopen it

File: config.yaml [hunk 1]
```diff
@@ -13,4 +13,18 @@ services:
...
+k8s-service-config:
+  interval: 30s
+  conditions:
+    - "[STATUS] == 200"
+
+exclude-suffix:
+  - primary
+  - canary
+
+k8s-cluster-mode: "out"
```
Comment (TwinProduction) [inline config.yaml:30]: Could you move these parameters under `kubernetes`?

i.e.

```yaml
kubernetes:
  cluster-mode: "out"
  auto-discover: true
  exclude-suffix:
    - primary
    - canary
  service-template:
    interval: 30s
    conditions:
      - "[STATUS] == 200"
```

**Review by TwinProduction — CHANGES_REQUESTED** [review]

**Comment by adarshk-rapido** [comment]
@TwinProduction we are running with these changes for the last couple of days at Rapido and looks good so far. 
Please take a look at the code change, meanwhile, I will add docs for these.

File: config.yaml (hunk 1 above)
Comment (adarshk-rapido) [reply config.yaml:30]: Sure will make this change.

Comment (adarshk-rapido) [reply config.yaml]: @TwinProduction Have made the changes.

**Comment by TwinProduction** [comment]
Great work @adarshk-rapido and thank you for the contribution!

I just tried it out locally and it's great.
There's a couple of changes I'll have to make to retain backward compatibility (i.e. if the `kubernetes` configuration is not defined, it'll panic), but besides from this, it looks great!

**Comment by TwinProduction** [comment]
Since I don't have write access to your fork, I'll push this to a branch, make a couple of changes to that branch and then push to master

**Comment by TwinProduction** [comment]
@adarshk-rapido The full list of changes can be found through the commits in #30, but here's the breaking ones from your initial implementation:
- Renamed `kubernetes.exclude-suffix` to `kubernetes.excluded-service-suffixes`
- Renamed `kubernetes.namespaces[].service-suffix` to `kubernetes.namespaces[].hostname-suffix`
- Renamed `kubernetes.namespaces[].health-api` to `kubernetes.namespaces[].target-path`

Breaking changes aside, I've...
- Added `kubernetes.namespaces[].excluded-services`
- Added a few tests
- Improved the documentation
- Specified that Kubernetes is in Alpha stage and may be subject to breaking changes for as long as it's in Alpha

It's up to you whether you decide to continue using your fork on your side or not, but I figured you'd want to know about these breaking changes if you did decide to move out from your branch.

**Comment by TwinProduction** [comment]
Released in [v1.4.0](https://github.com/TwinProduction/gatus/releases/tag/v1.4.0)

**Comment by adarshk-rapido** [comment]
Thanks for this awesome project @TwinProduction and glad we could add a little value to it from our side.

**Pull request merged**